ALAuto.py

```diff
@@ -200,11 +202,31 @@ def print_cycle_stats(self):
 
 try:
     while True:
-        Utils.update_screen()
+        Utils.wait_update_screen(1)
 
         # temporal solution to event alerts
-        if not Utils.find("menu/button_battle"):
-            Utils.touch_randomly(Region(54, 57, 67, 67))
+        # if not Utils.find("menu/button_battle"):
+        #     Utils.touch_randomly(Region(54, 57, 67, 67))
+        #     Utils.script_sleep(1)
+        #     continue
+        if Utils.find("menu/item_found"):
+            Logger.log_msg("Found item message.")
+            Utils.find_and_touch("menu/tap_to_continue")
+            Utils.script_sleep(1)
+            continue
+        if Utils.find("menu/home_button"):
+            Logger.log_msg("Found home button")
+            Utils.find_and_touch("menu/home_button")
+            Utils.script_sleep(1)
+            continue
+        if Utils.find("menu/announcement"):
+            Logger.log_msg("Found Announcement Window")
+            Utils.find_and_touch("menu/announcement")
+            Utils.script_sleep(1)
+            continue
+        if Utils.find("menu/alert_info"):
+            Logger.log_msg("Found alert.")
+            Utils.find_and_touch("menu/alert_close")
```

Comment:
Why is this necessary, does not the current implementation already go back to the main menu?

Reply:
The current implementation just clicks on a point in the upper left if it does not detect the button_battle asset , I think it quite incomplete and opens the window for several failures.

Reply:
For example, if this popup sho up, the current implementation will not fix, already woke up a few times and saw the bot stuck in this message.

![](https://i.imgur.com/26BplCM.png)